src/main/scala/com/cognite/sdk/scala/v1/functions.scala

```diff
@@ -0,0 +1,80 @@
+package com.cognite.sdk.scala.v1
+
+import java.time.Instant
+import com.cognite.sdk.scala.common._
+import io.circe.Json
+
+final case class FunctionError(
+    message: Option[String] = None,
+    trace: Option[String] = None
+)
+
+final case class Function(
+    id: Option[Long] = None,
+    name: String = "",
+    fileId: Long = 0,
+    owner: Option[String] = None,
+    description: Option[String] = None,
+    apiKey: Option[String] = None,
+    secrets: Option[Map[String, String]] = None,
+    createdTime: Option[Instant] = None,
+    status: Option[String] = None,
+    externalId: Option[String] = None,
+    error: Option[FunctionError] = None
+)
```

Comment:
Add
```scala
extends WithId[Long]
    with WithExternalId
    with WithCreatedTime
```

Reply:
Could not use WithCreatedTime since createdTime is option. But added the others, however, WithId[Option[Long]]